### PR TITLE
[skip e2e] Move codecov as separate job

### PR DESCRIPTION
**Pull request by jennyli-z**
Signed-off-by: Jenny Li <jing.li@zilliz.com>
/kind improvement

### CODE REVIEW

**Comment by jennyli-z** [comment]
rerun ut

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -85,21 +85,41 @@ jobs:
...
+        run: ls -lah
       - name: Upload coverage to Codecov
         if: "github.repository == 'milvus-io/milvus'"
         uses: codecov/codecov-action@v3.1.1
         id: upload_cov
         with:
           token: ${{ secrets.CODECOV_TOKEN }}
           files: ./go_coverage.txt,./lcov_output.info
-          name: ubuntu-${{ matrix.ubuntu }}-unittests
+          name: ubuntu-20.04-unittests
```
Comment (congqixia) [inline .github/workflows/main.yaml:123]: do we need to hard code this os version here? does using `${{ matrix.ubuntu }}` cause any problem?

Comment (jennyli-z) [reply .github/workflows/main.yaml:123]: the {{ matrix.ubuntu }} variable can only be used in the job UT, not able to be used in upload code coverage.

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**